flink-autoscaler/src/main/java/org/apache/flink/autoscaler/state/AutoScalerStateStore.java

```diff
@@ -41,23 +40,22 @@ void storeScalingHistory(
             Context jobContext, Map<JobVertexID, SortedMap<Instant, ScalingSummary>> scalingHistory)
             throws Exception;
 
-    Optional<Map<JobVertexID, SortedMap<Instant, ScalingSummary>>> getScalingHistory(
-            Context jobContext) throws Exception;
+    Map<JobVertexID, SortedMap<Instant, ScalingSummary>> getScalingHistory(Context jobContext)
```

Comment:
Could we mark these get interfaces as `@Nonnull`?

It's useful to let other developers know this limitation if they want to implement new state store. Otherwise, autoscaler will have a series of NullPointerException.

Reply:
Sure, that makes sense.